pi/heartbeat.py

```diff
@@ -192,6 +194,10 @@ def send_heartbeat(flic_last_seen_secs, flic_last_ping_secs, system_id):
                 num_secs_darkstat = parse_flic_last_seen_from_darkstat_html(html, FLIC_MAC_ADDRESS)
                 num_secs_ping = (datetime.datetime.now() - last_ping).total_seconds()
                 system_ok = send_heartbeat(num_secs_darkstat, num_secs_ping, system_id)
+                if not system_ok:
+                    # retry heartbeat request once
+                    # if the issue was a timeout error and the retry works, this will avoid turning off the Flic hub
```

Comment:
What do you mean by "turning off the Flic hub"?

Reply:
if `system_ok` is false, then later on in the run loop (in the `finally` block) the RPi turns off the relay that controls power to the Flic hub. This is so that the buttons don't flash green while there is something wrong with the communication between the Brave Hub and the server (ie. maybe the server is down, or the Brave Hub has a network connectivity problem). 

however, if the problem is just a one-off timeout in `send_heartbeat` then it would be better to avoid power cycling the Flic hub for no reason. the Slack thread linked from the ClickUp task has detailed logs of one of the 'flicker' events that show that this does happen with the current version of `heartbeat.py`

Reply:
Thanks for explaining!